apps/fileworker/src/lib/api.ts

```diff
@@ -77,18 +77,31 @@ export const router = new Hono<HonoApp>()
 				filename: z.string(),
 			}),
 		),
+		zValidator(
+			'query',
+			z.object({
+				expiration_ttl: z.coerce
+					.number()
+					.min(1)
+					.default(60 * 60 * 24)
+					.describe('number of seconds the file is valid for (defaults to 1 day)'),
+			}),
+		),
 		async (c) => {
 			const { filename } = c.req.valid('param')
+			const { expiration_ttl } = c.req.valid('query')
 			const file = await c.req.blob()
 			const { file_id } = await c.var.store.insertFile({
 				filename,
-				expires_on: datePlus('1 day'),
+				expires_on: new Date(Date.now() + expiration_ttl * 1000),
```

Comment:
Would we want to return this in the API and show it in the Front-end?

Reply:
oh yeah that sounds good! Added!